src/master/develop/transactions.md

```diff
@@ -1,6 +1,89 @@
 # Transactions
+Immudb supports transactions both on key-value and SQL level, but interactive transactions are supported only on SQL with the exception of `execAll` method, that provides some additional properties.
 
-`GetAll`, `SetAll` and `ExecAll` are the foundation of transactions in immudb. They allow the execution of a group of commands in a single step, with two important guarantees:
+## SQL interactive transactions
+Interactive transactions are a way to execute multiple SQL statements in a single transaction having the possibility to mix application logic with SQL statements.
+In order to create a transaction, you must call the `NewTx()` method on the client instance. The resulting object is a transaction object that can be used to execute multiple SQL statements, queries, commit or rollback.
+Following there are methods exposed by the transaction object:
+```
+Commit() CommittedSQLTx, error
+Rollback() error
+SQLExec(sql, params) error
+SQLQuery(sql, params) SQLQueryResult, error
+```
+It's possible to rollback a transaction by calling the `Rollback()` method. In this case, the transaction object is no longer valid and should not be used anymore.
+To commit a transaction, you must call the `Commit()` method.
+
+> **Note**: At the moment immudb support only 1 read-write transaction at a time, so it's up the application to ensure that only one read-write transaction is open at a time, or to handle read coflict error.
```

Comment:
TODO: What would be the result of a transaction conflict? Can we have some snippet that shows how to determine if there's a conflict?